app/controllers/downvotes_controller.rb

```diff
@@ -0,0 +1,29 @@
+class DownvotesController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
```

Comment:
Basically everything in this controller is a perfect candidate for a refactor into a PORO (plain old ruby object). Everything that this controller is doing can basically be boiled down into handling the logic of 'making a downvote'.

Here is an example of how I would recommend refactoring this controller. (You'll need to figure out how to write the code to make this work.)

```
def create
    VoteManager.new(current_user, review).downvote
    redirect_to sitcom
end
```

Your VoteManager object would get initialized with the user and the review object and then have the `downvote` method called on it. I'm suggesting a `VoteManager` object because for your UpvotesController you'd probably want to have a `upvote` method to call on that object since I'd guess that they both share some of the same concerns or logic.

I suggest initializing it with the user and review because you are then passing it all the data (user and review) and information about what you want done with that date (calling the method downvote). You'd probably start your `VoteManager` class like so:

````
class VoteManager
    attr_reader :user, :review

    def initialize(user, review)
        @user = user
        @review = review
    end

    def downvote
        ...<your code>
    end
end